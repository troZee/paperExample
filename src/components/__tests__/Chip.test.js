import * as React from 'react';
import renderer from 'react-test-renderer';
import Chip from '../Chip.tsx';

it('renders chip with onPress', () => {
  const tree = renderer
    .create(<Chip onPress={() => {}}>Example Chip</Chip>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders chip with icon', () => {
  const tree = renderer
    .create(<Chip icon="information">Example Chip</Chip>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders chip with close button', () => {
  const tree = renderer
    .create(
      <Chip icon="information" onClose={() => {}}>
        Example Chip
      </Chip>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders outlined disabled chip', () => {
  const tree = renderer
    .create(
      <Chip mode="outlined" disabled>
        Example Chip
      </Chip>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders selected chip', () => {
  const tree = renderer.create(<Chip selected>Example Chip</Chip>).toJSON();

  expect(tree).toMatchSnapshot();
});
