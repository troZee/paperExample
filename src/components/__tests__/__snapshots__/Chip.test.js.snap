// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chip with close button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "marginLeft": 4,
                "marginRight": 4,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 4,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 16,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders chip with icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "marginLeft": 4,
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders chip with onPress 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "marginLeft": 8,
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders outlined disabled chip 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={
      Array [
        "disabled",
      ]
    }
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.26)",
                "fontFamily": "System",
                "fontWeight": "400",
                "marginLeft": 8,
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders selected chip 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(188, 188, 188)",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={
      Array [
        "selected",
      ]
    }
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 4,
            },
            null,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 18,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "marginLeft": 4,
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;
